distributed/shuffle/_shuffle.py

```diff
@@ -367,9 +367,9 @@ def split_by_partition(
     Split data into many arrow batches, partitioned by final partition
     """
     import numpy as np
+    import pyarrow as pa
 
-    partitions = t.select([column]).to_pandas()[column].unique()
-    partitions.sort()
+    partitions = np.array(pa.compute.unique(t[column]).sort())
```

Comment:
Can we avoid the remainder of the `np` conversion as well by leveraging `pa.compute.equal` and `pa.compute.indices_nonzero`?

Reply:
Alternatively, maybe combining `value_counts`, `sort`, and a `cumsum` would be even more efficient?